vector/src/main/java/im/vector/activity/VectorRoomCreationActivity.java

```diff
@@ -333,26 +325,27 @@ public static String isDirectChatRoomAlreadyExist(String aUserId, MXSession mSes
                 if (directChatRoomsDict.containsKey(aUserId)) {
                     ArrayList<String> roomIdsList = new ArrayList<>(directChatRoomsDict.get(aUserId));
 
-                    if (0 != roomIdsList.size()) {
+                    if (!roomIdsList.isEmpty()) {
                         for (String roomId : roomIdsList) {
                             Room room = mSession.getDataHandler().getRoom(roomId, false);
-
                             // check if the room is already initialized
-                            if ((null != room) && room.isReady() && !room.isInvited() && !room.isLeaving()) {
-                                //dinsic: if the member is not already in matrix and just invited he's not active but
-                                // the room can be considered as ok
-                                if (!MXSession.isUserId(aUserId)){
-                                    return roomId;
-                                }
-                                else {
-                                    // test if the member did not leave the room
-                                    Collection<RoomMember> members = room.getActiveMembers();
-
-                                    for (RoomMember member : members) {
-                                        if (TextUtils.equals(member.getUserId(), aUserId)) {
-                                            Log.d(LOG_TAG, "## isDirectChatRoomAlreadyExist(): for user=" + aUserId + " roomFound=" + roomId);
-                                            return roomId;
+                            // dinsic: if the member is not already in matrix and just invited he's not active but
+                            // the room can be considered as ok
+                            if ((null != room) && !room.isLeaving()) {
```

Comment:
You forgot again `room.isReady()` condition

Reply:
No, if I add room.isReady() we will have a threads concurrency issue. In this case, the pending invitation will not be accepted and a new direct chat will be created.